ovos_utils/sound/__init__.py

```diff
@@ -32,110 +33,86 @@ def _get_pulse_environment(config):
-        read_mycroft_config().get('sounds', {}).get('end_listening'))
+def play_listening_sound(bus=None):
+    """Audibly indicate speech recording started."""
+    return _play_specific_sound('start_listening', bus=bus)
 
-    if not audio_file:
-        LOG.debug("Could not find 'end_listening' audio file!")
-        return
 
-    process = play_audio(audio_file)
-    if not process:
-        LOG.warning("Unable to play 'end_listening' audio file!")
-    return process
+def play_end_listening_sound(bus=None):
+    """Audibly indicate speech recording is no longer happening."""
+    return _play_specific_sound('end_listening', bus=bus)
 
 
-def play_error_sound():
+def play_error_sound(bus=None):
     """Audibly indicate a failed request.
 
     This method plays a error sound to signal an error that does not
     require a verbal response. This is intended to provide simple feedback
     to the user that their request was NOT handled successfully.
     """
-    audio_file = resolve_resource_file(
-        read_mycroft_config().get('sounds', {}).get('error'))
-
-    if not audio_file:
-        LOG.warning("Could not find 'error' audio file!")
-        return
-
-    process = play_audio(audio_file)
-    if not process:
-        LOG.warning("Unable to play 'error' audio file!")
-    return process
+    return _play_specific_sound('error', bus=bus)
 
 
 def _find_player(uri):
     _, ext = os.path.splitext(uri)
 
     # scan installed executables that can handle playback
     sox_play = find_executable("play")
-    pulse_play = find_executable("paplay")
-    alsa_play = find_executable("aplay")
-    mpg123_play = find_executable("mpg123")
-    ogg123_play = find_executable("ogg123")
-
-    player = None
     # sox should handle almost every format, but fails in some urls
     if sox_play:
-        player = sox_play + f" --type {ext} %1"
+        return sox_play + f" --type {ext} %1"
     # determine best available player
-    else:
-        if "ogg" in ext and ogg123_play:
-            player = ogg123_play + " -q %1"
-        # wav file
-        if 'wav' in ext:
-            if pulse_play:
-                player = pulse_play + " %1"
-            elif alsa_play:
-                player = alsa_play + " %1"
-        # guess mp3
-        elif mpg123_play:
-            player = mpg123_play + " %1"
-    return player
-
-
-def play_audio(uri, play_cmd=None, environment=None):
+    ogg123_play = find_executable("ogg123")
+    if "ogg" in ext and ogg123_play:
+        return ogg123_play + " -q %1"
+    # wav file
+    if 'wav' in ext:
+        pulse_play = find_executable("paplay")
+        if pulse_play:
+            return pulse_play + " %1"
+        alsa_play = find_executable("aplay")
+        if alsa_play:
+            return alsa_play + " %1"
+    # guess mp3
+    mpg123_play = find_executable("mpg123")
+    if mpg123_play:
+        return mpg123_play + " %1"
+    LOG.error("Can't find player for: %s", uri)
+    return None
+
```

Comment:
needed here, or could we split for scope? this is a different sort of testing challenge

Reply:
indeed this needs refactoring too, this results from the merging of similar utils to play audio, the ones here and later the ones from mycroft.util module

below we have play_ogg/mp3/wav methods that take the mycroft.conf into account, the behavior above determines best player but should only be used when play command is not set

perhaps here we should determine file extension only and this logic should be moved into the respective player methods, then called here

Reply:
let's keep this all to a follow up PR, @goldyfruit just noted some pulseaudio related issues aswel so this will need more work and this PR is already ballooning out of scope